File: topcoder/src/main/java/topcoder/java/SRM419_2_1.java
package topcoder.java;

public class SRM419_2_1 {
    public static int getPerimiter(int[] a) {
        
        int t = a[0] + a[a.length - 1] + a.length * 2;
        for(int i = 1; i < a.length; i++) {
            t += Math.abs(a[i] - a[i - 1]);
        }
        return t;
    }
    
}

File: topcoder/src/main/scala/topcoder/SRM419_2_1.scala
package topcoder

object SRM419_2_1 {
  def getPerimiter(a: Array[Int]) = {
    a.size * 2 + a(0) + a.last + 
      (a zip a.tail).map(i => math.abs(i._1 - i._2)).sum
  }
}